src/lib/test-fd-lock.ts

```diff
@@ -0,0 +1,16 @@
+import fs from 'fs';
+import path from 'path';
+import lock from 'fd-lock';
+
+async function testFdLock(dir: string) {
```

Comment:
Important for doing our tests we are using the temporary directory `dir` @tegefaulkes.